### PR TITLE
[v2] Project Add Route

### LINKED ISSUE #110: [v2] Project Routes
The following project routes need to be added:
- [x] `/project/` (GET): Returns a list of all (approved) projects. (#114)
- [x]  `/project/add/` (POST): Registers a new project. (#112)
- [x] `/project/details/` (POST): Returns the details of a particular project. (project id provided in JSON body) (#114)
- [x] `/project/update/` (PUT): Update a particular project. (project id and details provided in the JSON body) (#115)

### PR DESCRIPTION
**Pull request by harshkhandeparkar**
#### Changes
- Added `/project/add/` (POST) project registration endpoint.
- Added tests for the same.

### CODE REVIEW

**Review by proffapt — DISMISSED** [review]
Thoda chhota karo, itna bada nahi padhna, dimag hang hojata hai

File: server/routes.go [hunk 1]
```diff
@@ -51,5 +51,11 @@ func getRoutes(app *middleware.App) []Route {
 			"/healthcheck/ping/",
 			controllers.Ping,
 		},
+		{
+			"Project Registration",
+			"POST",
+			"/project/add/",
+			middleware.WithLogin(middleware.WrapApp(app, controllers.RegisterProject)),
```
Comment (rajivharlalka) [inline server/routes.go:58]: does the route need to have the /add/ ? Ig `POST /project/` itself refers to adding a project

File: controllers/project_reg.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+			"%s %s %s\n%s %s",
+			r.Method,
+			r.RequestURI,
+			"POSSIBLE SESSION HIJACKING.",
+			fmt.Sprintf("JWT Username: %s", login_username),
+			fmt.Sprintf("Given Username: %s", reqFields.MentorUsername),
+		)
+
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprint(w, "Login username and mentor username do not match.")
```
Comment (rajivharlalka) [inline controllers/project_reg.go:60]: use function to return error

File: controllers/project_reg.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		SecondaryMentor: secondaryMentor,
+	})
+
+	if tx.Error != nil {
+		utils.LogErrAndRespond(r, w, err, "Error adding the project in the database.", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprintf(w, "Success.")
```
Comment (rajivharlalka) [inline controllers/project_reg.go:146]: Ig returning a json object should be preferred

File: server/routes.go (hunk 1 above)
Comment (harshkhandeparkar) [reply server/routes.go:58]: Backend v1 has `/add`

File: controllers/project_reg.go (hunk 2 above)
Comment (harshkhandeparkar) [reply controllers/project_reg.go]: done

File: controllers/project_reg.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+		log.Warn().Msgf(
+			"%s %s %s\n%s %s",
+			r.Method,
+			r.RequestURI,
+			"POSSIBLE SESSION HIJACKING.",
+			fmt.Sprintf("JWT Username: %s", login_username),
+			fmt.Sprintf("Given Username: %s", reqFields.MentorUsername),
+		)
+
+		utils.LogWarnAndRespond(r, w, "Login username and mentor username do not match.", http.StatusUnauthorized)
```
Comment (rajivharlalka) [inline controllers/project_reg.go:58]:
```suggestion
	utils.LogWarnAndRespond(r, w, "Login username and mentor username do not match.", http.StatusUnauthorized)
```

Ig the function logs too

Comment (rajivharlalka) [inline controllers/project_reg.go:58]:
```suggestion
		utils.LogWarnAndRespond(r, w, "Login username and mentor username do not match.", http.StatusUnauthorized)
```

ig the function logs too

**Review by rajivharlalka — CHANGES_REQUESTED** [review]

File: server/routes.go (hunk 1 above)
Comment (rajivharlalka) [reply server/routes.go:58]: yes

Comment (rajivharlalka) [reply server/routes.go:58]: as long as the method are different, they dont overlap

**Review by rajivharlalka — APPROVED** [review]
LGTM

**Pull request merged**